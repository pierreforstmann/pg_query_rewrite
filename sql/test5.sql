drop extension if exists pg_query_rewrite;
--
create extension pg_query_rewrite;
select pgqr_truncate();
--
drop table if exists t;
create table t(x int, y int);
--
insert into t values(1,2);
--
select * from t;
--
select pgqr_add_rule('delete from t;','delete from t where 1=0;');
--
delete from t;
--
select * from t;
--
select pgqr_rules();
