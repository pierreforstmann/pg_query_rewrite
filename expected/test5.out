drop extension if exists pg_query_rewrite;
--
create extension pg_query_rewrite;
select pgqr_truncate_rule();
 pgqr_truncate_rule 
--------------------
 t
(1 row)

--
drop table if exists t;
create table t(x int, y int);
--
insert into t values(1,2);
--
select * from t;
 x | y 
---+---
 1 | 2
(1 row)

--
select pgqr_add_rule('delete from t;','delete from t where 1=0;');
 pgqr_add_rule 
---------------
 t
(1 row)

--
delete from t;
--
select * from t;
 x | y 
---+---
 1 | 2
(1 row)

--
select pgqr_rules();
                                               pgqr_rules                                               
--------------------------------------------------------------------------------------------------------
 (datname=contrib_regression,"source=delete from t;","target=delete from t where 1=0;",rewrite_count=1)
 (datname=NULL,source=NULL,target=NULL,rewrite_count=0)
 (datname=NULL,source=NULL,target=NULL,rewrite_count=0)
 (datname=NULL,source=NULL,target=NULL,rewrite_count=0)
 (datname=NULL,source=NULL,target=NULL,rewrite_count=0)
 (datname=NULL,source=NULL,target=NULL,rewrite_count=0)
 (datname=NULL,source=NULL,target=NULL,rewrite_count=0)
 (datname=NULL,source=NULL,target=NULL,rewrite_count=0)
 (datname=NULL,source=NULL,target=NULL,rewrite_count=0)
 (datname=NULL,source=NULL,target=NULL,rewrite_count=0)
(10 rows)

